### PR TITLE
Fixed #58

### LINKED ISSUE #58: Add outer join
```scala
def joinOuter[U](ts: TimeSeries[U], defaultLeft: V, defaultRight: U): TimeSeries[(V, U)]
```

This join will work similar to SQL full outer join

### PR DESCRIPTION
**Pull request by wenta**

### CODE REVIEW

File: src/main/scala/carldata/series/TimeSeries.scala [hunk 1]
```diff
@@ -470,5 +470,30 @@ case class TimeSeries[V](idx: Vector[LocalDateTime], ds: Vector[V]) {
     new TimeSeries(joinR(dataPoints, ts.dataPoints))
   }
 
+  /** Outer join. If one of the series doesn't have a data point then put default(left or right) value. */
+  def joinOuter[U](ts: TimeSeries[U], defaultLeft: V, defaultRight: U): TimeSeries[(V, U)] = {
+    val builder: mutable.ListBuffer[(LocalDateTime, (V, U))] = ListBuffer()
+
+    @tailrec def joinR(dp1: Vector[(LocalDateTime, V)],
+                       dp2: Vector[(LocalDateTime, U)]): Seq[(LocalDateTime, (V, U))] = {
+      if (dp1.isEmpty | dp2.isEmpty) builder
```
Comment (klangner) [inline src/main/scala/carldata/series/TimeSeries.scala]: should be '||'

File: src/main/scala/carldata/series/TimeSeries.scala [hunk 2]
```diff
@@ -470,5 +470,30 @@ case class TimeSeries[V](idx: Vector[LocalDateTime], ds: Vector[V]) {
     new TimeSeries(joinR(dataPoints, ts.dataPoints))
   }
 
+  /** Outer join. If one of the series doesn't have a data point then put default(left or right) value. */
+  def joinOuter[U](ts: TimeSeries[U], defaultLeft: V, defaultRight: U): TimeSeries[(V, U)] = {
+    val builder: mutable.ListBuffer[(LocalDateTime, (V, U))] = ListBuffer()
+
+    @tailrec def joinR(dp1: Vector[(LocalDateTime, V)],
```
Comment (klangner) [inline src/main/scala/carldata/series/TimeSeries.scala]: I think it is better to name params xs and ys instead of dp1 and dp2
And then x instead of p1.

File: src/test/scala/carldata/series/TimeSeriesTest.scala [hunk 3]
```diff
@@ -309,6 +309,19 @@ class TimeSeriesTest extends FlatSpec with Matchers {
     series1.joinLeft(series2, 0) shouldBe expected
   }
 
+  it should "outer join 2 series" in {
```
Comment (klangner) [inline src/test/scala/carldata/series/TimeSeriesTest.scala:312]: Could you add test where one series is empty?

**Review by klangner — CHANGES_REQUESTED** [review]

**Review by klangner — APPROVED** [review]
